src/components/rangeslider/draw.js

```diff
@@ -0,0 +1,424 @@
+        }
+    });
+
+    // for all present range slides
+    rangeSliders.each(function(axisOpts) {
+        var rangeSlider = d3.select(this),
+            opts = axisOpts[constants.name];
+
+        // update range slider dimensions
+
+        var margin = fullLayout.margin,
+            graphSize = fullLayout._size,
+            domain = axisOpts.domain;
+
+        opts._width = graphSize.w * (domain[1] - domain[0]);
+        opts._height = (fullLayout.height - margin.b - margin.t) * opts.thickness;
+        opts._offsetShift = Math.floor(opts.borderwidth / 2);
+
+        var x = margin.l + (graphSize.w * domain[0]),
+            y = fullLayout.height - opts._height - margin.b;
+
+        rangeSlider.attr('transform', 'translate(' + x + ',' + y + ')');
+
+        // update inner nodes
+
+        rangeSlider
+            .call(drawBg, gd, axisOpts, opts)
+            .call(drawRangePlot, gd, axisOpts, opts)
+            .call(drawMasks, gd, axisOpts, opts)
+            .call(drawSlideBox, gd, axisOpts, opts)
+            .call(drawGrabbers, gd, axisOpts, opts);
+
+        // update data <--> pixel coordinate conversion methods
+
+        var range0 = opts.range[0],
+            range1 = opts.range[1],
+            dist = range1 - range0;
+
+        opts.p2d = function(v) {
+            return (v / opts._width) * dist + range0;
+        };
+
+        opts.d2p = function(v) {
+            return (v - range0) / dist * opts._width;
+        };
+
+        // setup drag element
+        setupDragElement(rangeSlider, gd, axisOpts, opts);
+
+        // update current range
+        setPixelRange(rangeSlider, gd, axisOpts, opts);
+
+        // update margins
+
+        var bb = axisOpts._boundingBox ? axisOpts._boundingBox.height : 0;
+
+        Plots.autoMargin(gd, constants.name + axisOpts._id, {
+            x: 0, y: 0, l: 0, r: 0, t: 0,
+            b: opts._height + fullLayout.margin.b + bb,
+            pad: 15 + opts._offsetShift * 2
+        });
+    });
+};
+
+function makeRangeSliderData(fullLayout) {
+    if(!fullLayout.xaxis) return [];
+    if(!fullLayout.xaxis[constants.name]) return [];
+    if(!fullLayout.xaxis[constants.name].visible) return [];
+    if(fullLayout._has('gl2d')) return [];
+
+    return [fullLayout.xaxis];
+}
+
+function setupDragElement(rangeSlider, gd, axisOpts, opts) {
+    var slideBox = rangeSlider.select('rect.' + constants.slideBoxClassName).node(),
+        grabAreaMin = rangeSlider.select('rect.' + constants.grabAreaMinClassName).node(),
+        grabAreaMax = rangeSlider.select('rect.' + constants.grabAreaMaxClassName).node();
+
+    rangeSlider.on('mousedown', function() {
+        var event = d3.event,
+            target = event.target,
+            startX = event.clientX,
+            offsetX = startX - rangeSlider.node().getBoundingClientRect().left,
+            minVal = opts.d2p(axisOpts.range[0]),
+            maxVal = opts.d2p(axisOpts.range[1]);
+
+        var dragCover = dragElement.coverSlip();
```

Comment:
Note: I didn't use the full [`dragElement`](https://github.com/plotly/plotly.js/blob/9b249c2f7facb1c323dd226f0b0dbdee91075b7e/src/components/dragelement/index.js#L51) routine because it slowed down the `requestAnimationFrame` calls considerably.